### PR TITLE
Levenshtein distance computation.

### PR DESCRIPTION
**Pull request by dkorolev**
Hi @mzhurovich ,

I coded this a while back in a clean branch, hoping the trick would help making ASCII-art-based `gnuplot` tests pass.

Propose we merge it into Bricks.

PTAL.

Thanks,
Dima

### CODE REVIEW

File: strings/distance.h [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+    }
+
+    size_t Compute() const { return Compute(la, lb); }
+  };
+  SlowImpl impl(a, b);
+  return impl.Compute();
+}
+
+// Returns the edit distance between two strings, assuming it is forbidden
+// to 'skew' the strings by more than `max_offset` apart while mathing them letter-by-letter.
```
Comment (mzhurovich) [inline strings/distance.h]: s/mathing/matching/

File: strings/distance.h [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+        return static_cast<size_t>(-1);
+      }
+    }
+    size_t Compute() const { return Compute(la, lb); }
+  };
+  FastImpl impl(a, b, max_offset);
+  return impl.Compute();
+}
+
+inline size_t FastEditDistance(const std::string& a, const std::string& b, int max_offset) {
```
Comment (mzhurovich) [inline strings/distance.h]: Do we really want to expose this kind of signature in this particular way?
Accidentally passed `max_offset = -1` will result in a really huge vector.

File: strings/distance.h (hunk 1 above)
Comment (dkorolev) [inline strings/distance.h]: :+1: fixed.

File: strings/distance.h (hunk 2 above)
Comment (dkorolev) [inline strings/distance.h]: Good point. Really huge vector is not a problem (`std::bad_alloc` will get thrown instantly), but why not train users to use Bricks like they use STL :-) fixed.

**Comment by dkorolev** [comment]
Thanks for the review, @mzhurovich, PTAL.

(Sorry, my OCD kicked in and I added a few coverage-related comments and did `make indent` in MMQ as well :-) )

Thanks!
Dima

**Comment by mzhurovich** [comment]
@dkorolev LGTM :+1:
Merging.

**Pull request merged**